ppr-ui/src/composables/mhrRegistration/useNewMhrRegistration.ts

```diff
@@ -0,0 +1,120 @@
+import { useGetters } from 'vuex-composition-helpers'
+import {
+  getMhrAttentionReferenceNum,
+  getMhrRegistrationDocumentId,
+  getMhrRegistrationHomeDescription, getMhrRegistrationHomeOwnerGroups, getMhrRegistrationLocation,
+  getMhrRegistrationSubmittingParty
+} from '@/store/getters'
+import {
+  MhrRegistrationDescriptionIF,
+  MhrRegistrationHomeLocationIF,
+  MhrRegistrationHomeOwnerGroupIF,
+  NewMhrRegistrationApiIF
+} from '@/interfaces'
+
+export const useNewMhrRegistration = () => {
+  const {
+    getMhrRegistrationHomeDescription,
+    getMhrRegistrationSubmittingParty,
+    getMhrRegistrationDocumentId,
+    getMhrAttentionReferenceNum,
+    getMhrRegistrationLocation,
+    getMhrRegistrationHomeOwnerGroups
+  } = useGetters<any>([
+    'getMhrRegistrationHomeDescription',
+    'getMhrRegistrationSubmittingParty',
+    'getMhrRegistrationDocumentId',
+    'getMhrAttentionReferenceNum',
+    'getMhrRegistrationLocation',
+    'getMhrRegistrationHomeOwnerGroups'
+  ])
+
+  function cleanEmpty<Type> (obj:Type): Type {
+    const newObj = {}
+    Object.keys(obj).forEach((key) => {
+      if (obj[key] !== null && typeof obj[key] === 'object') newObj[key] = cleanEmpty(obj[key])
+      else if (!!obj[key] || obj[key] === 0) newObj[key] = obj[key]
+    })
+    return newObj as Type
+  }
```

Comment:
Why a function vs a function expression here?

Also a function like this deserves some comments and explanation to help other developers understand what this function is doing. 

Please also break the conditional blocks to make for easier readability using `{}`

Reply:
Because I couldn't figure out the syntax of function expression with type template...

will add some comments.